src/common/options.cc

```diff
@@ -7400,6 +7400,20 @@ std::vector<Option> get_mds_options() {
     .set_description("target maximum memory usage of MDS cache")
     .set_long_description("This sets a target maximum memory usage of the MDS cache and is the primary tunable to limit the MDS memory usage. The MDS will try to stay under a reservation of this limit (by default 95%; 1 - mds_cache_reservation) by trimming unused metadata in its cache and recalling cached items in the client caches. It is possible for the MDS to exceed this limit due to slow recall from clients. The mds_health_cache_threshold (150%) sets a cache full threshold for when the MDS signals a cluster health warning."),
 
+    Option("mds_cache_autotune", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
+    .set_default(false)
+    .set_description("Enable autotuning of mds_cache_memory_limit")
+    .set_long_description("When set, mds_cache_memory_limit is automatically tuned as per the rss usage periodically whose time interval is defined by mds_cache_resize_interval."),
+
+    Option("mds_memory_target", Option::TYPE_SIZE, Option::LEVEL_ADVANCED)
+    .set_default(1*(1LL<<30))
+    .set_description("Upper bound for the mds_cache_memory_limit")
+    .set_long_description("When cache autotuning is enabled, sets an upper bound for mds_cache_memory_limit" ),
```

Comment:
`mds_memory_target` will still be used to set the default value for the `mds_cache_memory_limit`, either when the MDS starts or when the config is changed at runtime.

Reply:
Ack.

Reply:
I would be careful about saying RSS here since that's not strictly true.  We are watching the mapped memory of the process, but there may be unmapped memory that the kernel has not yet reclaimed for some reason.